### DISCUSSION TITLE
Using replace function from useFieldArray sets form isDirty to true

### DISCUSSION THREAD

**Discussion by damien-duignan** [react-hook-form/react-hook-form#8223] [Q&A] [answered]
We have a form which can be edited, with the input fields populated with data coming from our API.

We populate some parts of this form using the `replace` function from `useFieldArray`, but using the `replace` function sets the form to `isDirty=true`. This doesn't make sense in this case as the user hasn't changed anything yet.

Here's a codesandbox to to show the issue - https://codesandbox.io/s/heuristic-phoebe-ldgfvc?file=/src/App.tsx (this is a simplified version for illustration purposes).

I would imagine that there could be an option to not mark these fields as dirty, similar to the options for `setValue`:
```
replace([{ name: 'test' }], { shouldDirty: false })
```

Does such an option make sense, or is it better to try populate the form with `defaultValues` in `useForm` hook?

**Comment by bluebill1049** [comment, marked as answer]
all field array actions will mark form state to dirty, there is no clear way for us to detect if it's triggered by the user or programmatically by the developer.

**Reply by damien-duignan** [reply]
Thanks for your reply.

Is there a reason that we can't pass an options object to the `replace` function? In the case described above, I could always pass `{ shouldDirty: false }` to make sure `isDirty` remains false.
```
replace([{ name: 'test' }], { shouldDirty: false })
```
Or has it just not been implemented yet?

**Reply by bluebill1049** [reply]
There is no such feature at the moment, you can send a feature request if you want and we will ping to project for upvotes.

**Reply by damien-duignan** [reply]
Ok, will do that. Thanks!

**Reply by ghost** [reply]
What's the update?  Is this feature coming or any other way to solve this?

**Reply by Moshyfawn** [reply]
We've considered adding the field state flags to field array methods like `replace` for v8, but it's not in development yet. We have very little time and a huge backlog

**Reply by Moshyfawn** [reply]
Related discussion #8309

**Reply by NarenderDhamarlapally** [reply]
reset({
  test: [{ value: "one" }, { value: "two" }]
});

this way correct or not

**Comment by NarenderDhamarlapally** [comment]
reset({
      test: [{ value: "one" }, { value: "two" }]
    });


this workaroud iswoking for me
